### PR TITLE
:bug: Fix FormDefinition admin for readonly access

### PR DESCRIPTION
**Pull request by stevenbal**
fixes: https://github.com/open-formulieren/open-forms/issues/998

Relatively minor issues, only occurred for staff users with readonly access

### CODE REVIEW

**Comment by sergei-maertens** [comment]
Some of these conflict with the work in #954 which has been pending a lot longer and is being finalized. Maybe wait until that is merged and then rebase for the remaining issues? :grimacing:

**Comment by sergei-maertens** [comment]
it's merged!

**Comment by stevenbal** [comment]
@sergei-maertens I rebased on master

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**
